src/main/java/frc/robot/subsystems/CoralRoller.java

```diff
@@ -124,11 +136,25 @@ public void periodic() {
 
   /** Updates and logs the current sensors states. */
   private void updateTelemetry() {
-    hasCoral = !beamBreak.get();
+    // hasCoral = !beamBreak.get();
     currentVelocity = motor.getVelocity().refresh().getValueAsDouble() * METERS_PER_REVOLUTION;
 
+    if (currentVelocity != 0) {
+      coralIntakeTime.start();
+    } else {
+      hasCoral = false;
+    }
+    if (coralIntakeTime.hasElapsed(CORAL_INTAKE_INTERVAL)) { // TODO: Find coral intake interval.
+      coralIntakeTime.stop();
+      coralIntakeTime.reset();
+      hasCoral = true;
+    }
```

Comment:
We'll need to comment this out when we run experiments to gather the additional current data in data logger.

The additional data logging is more generally useful, so it would be worth adding them in a separate PR we could merge while we work out an heuristic for coral detection.